src/sources/soundsourcem4a.cpp

```diff
@@ -506,6 +519,58 @@ void SoundSourceM4A::restartDecoding(MP4SampleId sampleBlockId) {
     m_sampleBuffer.clear();
 }
 
+int SoundSourceM4A::seekAndReadSampleBlock(
+        int sampleBlockId,
+        uint8_t** ppBytes,
+        uint32_t* pNumBytes) {
+    DEBUG_ASSERT(sampleBlockId > 0 && sampleBlockId <= m_pAvStream->nb_index_entries);
+
+    int error = 0;
+    char error_string[AV_ERROR_MAX_STRING_SIZE];
+
+    if (m_nextSampleBlockId != sampleBlockId) {
+        error = av_seek_frame(
+                m_pAvInputFormatContext,
+                m_pAvStream->index,
+                m_pAvStream->index_entries[sampleBlockId - 1].timestamp,
+                SEEK_SET);
+
+        if (error < 0) {
+            av_make_error_string(error_string, AV_ERROR_MAX_STRING_SIZE, error);
+            qWarning() << "Could not seek frame:" << error_string;
+            return error;
+        }
+    }
+
+    // create an empty package
+    AVPacket packet;
+    av_init_packet(&packet);
+    packet.data = nullptr;
+    packet.size = 0;
+
+    m_nextSampleBlockId = kSampleBlockIdInvalid;
+
+    error = av_read_frame(m_pAvInputFormatContext, &packet);
+    if (error < 0) {
+        av_make_error_string(error_string, AV_ERROR_MAX_STRING_SIZE, error);
+        qWarning() << "Could not read frame:" << error_string;
+    } else if (packet.size <= static_cast<int>(*pNumBytes)) {
+        memcpy(*ppBytes, packet.data, packet.size);
+        m_nextSampleBlockId = sampleBlockId + 1;
+        DEBUG_ASSERT(packet.pos == m_pAvStream->index_entries[sampleBlockId - 1].pos);
+    } else {
+        error = AVERROR(ENOMEM);
+    }
+    if (sampleBlockId == kSampleBlockIdMin) {
+        // first block contains extra data that make libfaad struggle.
+        *pNumBytes = 6;
```

Comment:
Why 6? And without any safety checks that packet.size is indeed >= 6 as expected?

Reply:
Maybe you have some insights. 
With libmp4 it is 
33 0 3 64 104 28
With ffmpeg
1 33 0 3 64 104 28 64 8 0 0 0 0 0 0 0 0 0 0 0 10 198 140 77 157 16 142 37 233 254

Reply:
Oh I have just noticed that his can be quite different.